js_modules/dagster-ui/packages/ui-core/src/search/useGlobalSearch.tsx

```diff
@@ -233,15 +233,14 @@ export const useGlobalSearch = () => {
     [],
   );
 
-  // If we already have WebWorkers set up, initialization is complete and this will be a no-op.
   const initialize = useCallback(() => {
-    if (!primarySearch.current) {
+    if (!primaryResult.data && !primaryResult.loading) {
```

Comment:
This gates `performPrimaryLazyQuery` more directly to avoid it being called more than once (even though apollo cache no-ops second calls while one is already in progress). I think this makes more sense than gating on `primarySearch.current` because we're not relying on the side effect that occurs in `onCompleted`